test/cqlpy/test_materialized_view.py

```diff
@@ -1569,3 +1569,10 @@ def test_views_with_future_tombstones(cql, test_keyspace):
             cql.execute(f'insert into {table} (a,b,c,d,e) values (3,4,5,6,7) using timestamp 18')
             assert [] == list(cql.execute(f'select * from {table}'))
             assert [] == list(cql.execute(f'select * from {mv}'))
+
+# Test that we can perform reads from the view in reverse order without crashing.
+# Reproduces issue https://github.com/scylladb/scylladb/issues/21354
+def test_reverse_read_from_view(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, 'a int PRIMARY KEY, b int') as table:
+        with new_materialized_view(cql, table, '*', 'b, a', 'a is not null and b is not null') as mv:
+            assert [] == list(cql.execute(f'select * from {mv} where b=1 order by a desc'))
```

Comment:
Excellent. But since we now discovered we have a special code path for reversed reading from a view, maybe it's worth checking that it doesn't just "not crash" but actually gives the result in reversed order. So I would add to this test inserting a few rows into the base partition, reading them in the original order without order by, again in the original order with `order by asc`, and check `order by desc` returns reversed order.

Reply:
Ok, I improved the test in the rebase